update controller.md (#665)
megaease/easegress: doc/reference/controllers.md

  kind: Retry
  maxAttempts: 3
``` 
In this case, if `proxy` returns non-empty results, then resilience retry reruns the `proxy` filter until `proxy` returns empty results or gets the max attempts. 

The `flow` also supports `namespace`, so the pipeline can support workflows that contain multiple requests and responses. 
```yaml
name: http-pipeline-example4
kind: Pipeline
flow: 
- filter: validator
  jumpIf:
    invalid: END 
- filter: requestBuilderFoo 
  namespace: foo 
- filter: proxyFoo 
  namespace: foo 
- filter: requestBuilderBar 
  namespace: bar 
- filter: proxyBar
  namespace: bar 
- filter: responseBuilder 

filters: 
- name: requestBuilder
  kind: RequestBuilder
  ...
... 
```
In this case, `requestBuilderFoo` creates a request in namespace `foo`, and `proxyFoo` sends `foo` request and puts the response into namespace `foo`. `requestBuilderBar` creates a request in namespace `bar` and `proxyBar` sends `bar` request and puts the response into namespace `bar`. Finally, `requestBuilder` create a response by using responses in namespace `foo` and `bar` and putting it into the default namespace. 

> If not set, the filter works in the default namespace `DEFAULT`.

> The `alias` in `flow` can give the filter an alias name.  So you can easily create different filters with the same spec. For example: 

Reviewer: ```suggestion
The `alias` in `flow` gives a filter an alias to help re-use the filter so that we can use the alias to distinguish each of its appearances in the flow.
```

| methods       | []string                                 | Methods to match, empty means to allow all methods                                                                                     | No       |
| headers       | [][httpserver.Header](#httpserverHeader) | Headers to match (the requests matching headers won't be put into cache)                                                               | No       |
| backend       | string                                   | backend name (pipeline name in static config, service name in mesh)                                                                    | Yes      |
| clientMaxBodySize | int64 | Max size of request body. the default value is 4MB. Requests with a body larger than this option are discarded.  When this option is set to `-1`, Easegress takes the request body as a stream and the body can be any size, but some features are not possible in this case. | No | 

> | matchAllHeader | bool | Match all headers that are defined in headers. | No |

Reviewer: ```suggestion
| matchAllHeader | bool | Match all headers that are defined in headers, default is `false`. | No |
```

| route53           | accessKeyId, secretAccessKey, awsProfile                            |
| vultr             | apiToken                                                            |

### resilience.Policy

| Name                 | Type   | Description    | Required |
| -------------------- | ------ | -------------- | -------- |
| name                 | string | Name of filter | Yes      |
| kind                 | string | Kind of filter | Yes      |

> | self-defining fields | -      | -              | -        |
> 
> Self-defining fields see [Retry](#retry) or [CircuitBreaker](#circuitBreaker). 

Reviewer: ```suggestion
| other kind specific fields of the policy kind | -      | -              | -        |
```

> #### Retry 
> Retry retries failed requests according to configured policy.

Reviewer: ```suggestion
#### Retry Policy
A retry policy configures how to retry a failed request.
```

| Name | Type | Description | Required |
|------|------|-------------|----------|
| maxAttempts | int | The maximum number of attempts (including the initial one). Default is 3 | No |
| waitDuration | string | The base wait duration between attempts. Default is 500ms | No |
| backOffPolicy | string  | The back-off policy for wait duration, could be `EXPONENTIAL` or `RANDOM` and the default is `RANDOM`. If configured as `EXPONENTIAL`, the base wait duration becomes 1.5 times larger after each failed attempt | No |
| randomizationFactor  | float64 | Randomization factor for actual wait duration, a number in interval `[0, 1]`, default is 0. The actual wait duration used is a random number in interval `[(base wait duration) * (1 - randomizationFactor),  (base wait duration) * (1 + randomizationFactor)]` | No |

> #### circuitBreaker

Reviewer: ```suggestion
#### Circuit Break Policy
```

When `CLOSED`, the CircuitBreaker uses a sliding window to store and aggregate the result of recent requests, the window can either be `COUNT_BASED` or `TIME_BASED`. The `COUNT_BASED` window aggregates the last N requests and the `TIME_BASED` window aggregates requests in the last N seconds, where N is the window size.

Below is an example configuration with both `COUNT_BASED` and `TIME_BASED` policies. Policy `circuit-breaker-example-count` short-circuits requests if more than half of recent requests failed.

Policy `circuit-breaker-example-time` short-circuits requests if more than 60% of recent requests failed.

> failed means that backend filter returns non-empty results. 

```yaml
kind: CircuitBreaker
name: circuit-breaker-example-count 
slidingWindowType: COUNT_BASED
failureRateThreshold: 50
slidingWindowSize: 100

---
kind: CircuitBreaker
name: circuit-breaker-example-time 
slidingWindowType: TIME_BASED
failureRateThreshold: 60
slidingWindowSize: 100

> ```

Reviewer: please refer https://github.com/megaease/easegress/blob/663a08344cc64fe6194c3fbe454c912e518a5a3a/doc/cookbook/resilience.md#circuit-break to update this part, and please make sure the kind should be `CircuitBreak` instead of `CircuitBreaker`.
